test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenTransactSpecs.java

```diff
@@ -432,6 +438,45 @@ public HapiApiSpec balancesChangeOnTokenTransfer() {
 				);
 	}
 
+	public HapiApiSpec uniqueTokenTransfers() {
+		return defaultHapiSpec("UniqueTokenTransfers")
+				.given(
+						cryptoCreate(FIRST_USER).balance(0L),
+						cryptoCreate(SECOND_USER).balance(0L),
+						cryptoCreate(TOKEN_TREASURY).balance(0L),
+						newKeyNamed("supplyKey"),
+						tokenCreate(A_TOKEN)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.initialSupply(0)
+								.supplyKey("supplyKey")
+								.treasury(TOKEN_TREASURY),
+						mintToken(A_TOKEN, List.of(ByteString.copyFromUtf8("memo"))),
+						tokenAssociate(FIRST_USER, A_TOKEN),
+						tokenAssociate(SECOND_USER, A_TOKEN)
+				).when(
+						cryptoTransfer(
+								moving(1, 1, A_TOKEN).betweenUnique(TOKEN_TREASURY, FIRST_USER)
```

Comment:
it seems that for naming it would be better if it was:
`movingUnique(1, A_TOKEN).between(TOKEN_TREASURY, FIRST_USER)`

Reply:
Confirmed and refactored for better meaning and usage